### PR TITLE
fix: fix the installation script for windows with the extension

### LINKED ISSUE #1097: Installation script not working on windows (wrong download url)
**Describe the bug**

**To Reproduce**
1. Run the installation curl script with MINGW64 on Windows
2. Download of the release is not working.

**Expected behavior**
Should download and unpack the correct file.

**Output**
```
$ curl -sSL https://router.apollo.dev/download/nix/latest | sh
Downloading release info for 'https://github.com/apollographql/router/releases'
Found router-0.9.1.exe-x86_64-windows.tar.gz
Downloading router from https://github.com/apollographql/router/releases/download/v0.9.1/router-0.9.1.exe-x86_64-windows.tar.gz
curl: (22) The requested URL returned error: 404
Failed to download https://github.com/apollographql/router/releases/download/v0.9.1/router-0.9.1.exe-x86_64-windows.tar.gz
```

**Desktop (please complete the following information):**
 - OS: Windows
 - Version 11

**Additional context**
Wrong URL: https://github.com/apollographql/router/releases/download/v0.9.1/router-0.9.1.exe-x86_64-windows.tar.gz
Correct URL: https://github.com/apollographql/router/releases/download/v0.9.1/router.exe-0.9.1-x86_64-windows.tar.gz

### PR DESCRIPTION
**Pull request by bnjjj**
close #1097

### CODE REVIEW

**Review by Geal — APPROVED** [review]

**Review by abernix — CHANGES_REQUESTED** [review]
I'm curious why the correct solution is not removing `.exe` from the name of the built tarball and aligning with what [we do on `rover`](https://github.com/apollographql/rover/releases/tag/v0.6.0), for example?

**Review by abernix — APPROVED** [review]

File: xtask/src/commands/package/mod.rs [hunk 1]
```diff
@@ -40,8 +40,7 @@ impl Package {
             self.output.to_owned()
         } else if self.output.is_dir() {
             self.output.join(format!(
-                "{}-{}-{}-{}.tar.gz",
-                RELEASE_BIN,
+                "router-{}-{}-{}.tar.gz",
```
Comment (o0Ignition0o) [inline xtask/src/commands/package/mod.rs:43]: not sure what is happenning so i ll ask away: is line 25 still correct? 

```
let release_path = TARGET_DIR.join("release").join(RELEASE_BIN);
```

**Review by o0Ignition0o — APPROVED** [review]
LGTM if tested on windows; I m not quite sure what is happening wrt router directory and binary and whatnot 🙈 😅

File: xtask/src/commands/package/mod.rs (hunk 1 above)
Comment (bnjjj) [reply xtask/src/commands/package/mod.rs:43]: It's good

**Pull request merged**